File: app/server/internal/product/product.go
package product

type Product struct {
	Id    uint64  `json:"id" db:"id"`
	Name  string  `json:"name" db:"name"`
	Price float32 `json:"price" db:"price"`
	Rank  int     `json:"rank" db:"rank"`
}

// Products can be used to inject abilities to Product slices like sorting, filtering, etc.
type Products []Product

File: app/client/src/types/Product.ts
import {Model} from "@vuex-orm/core";

export default class Product extends Model {
    static entity = "products";

    id!: number;
    name!: string;
    price!: number;
    rank!: number;

    static fields() {
        return {
            id: this.number(0),
            name: this.string(""),
            price: this.number(0),
            rank: this.number(0),
        };
    }
}